### PR TITLE
MxVideoPresenter functions

### PR DESCRIPTION
**Pull request by MishaProductions**
TODO: match MxVideoPresenter::Destroy

### CODE REVIEW

File: LEGO1/mxrect32.h [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #ifndef MXRECT32_H
 #define MXRECT32_H
 
+#define MxRect32GetWidth(rect) (rect.m_right - rect.m_left) + 1
+#define MxRect32GetHeight(rect) (rect.m_bottom - rect.m_top) + 1
+
```
Comment (disinvite) [inline LEGO1/mxrect32.h:6]: Does it affect the match percentage if these are inline getter methods instead?

Comment (MishaProductions) [reply LEGO1/mxrect32.h:6]: It shouldn't, but that's how microsoft does the rectangle width/height macros, so I assumed that's how mindscape did

Comment (MishaProductions) [reply LEGO1/mxrect32.h:6]: Nope, it doesn't affect the percentage, but it's cleaner.

**Comment by MishaProductions** [comment]
I don't really know how to make it match

**Comment by foxtacles** [comment]
Got the match up to ~88%. The rest *might* be compiler randomness but I think there's also a chance a minor piece needs to be changed, however I can't think of any.

Changes that were made:

* `m_unk58` is `LPDIRECTDRAWSURFACE`
* `UnkStruct` (previously `unknown_meta_struct`) seems to have a virtual table according to this function, so I added a virtual destructor for now. This structure still needs more investigation.
* Reordered some instructions

**Pull request merged**